datafusion/core/src/execution/context.rs

```diff
@@ -1510,6 +1515,27 @@ impl SessionState {
         Ok(statement)
     }
 
+    /// Convert a SQL string into an AST Statement
+    pub fn sql_to_statement_with_dialect(
```

Comment:
I think with this change, it will mean that `sql_to_statement` will effectively ignore `ConfigOptions::sql_parser::dialect` which seems confusing. 

Is there a need for a separate  "sql_to_statement_with_dialect" -- rather than changing  `sql_to_statement` to use the configured dialect ?